backtracking/rat_maze.cpp

```diff
@@ -42,21 +72,31 @@ int solveMaze(int currposrow, int currposcol, int maze[size][size],
         return 0;
     }
 }
+}  // namespace rat_maze
+}  // namespace backtracking
 
-int main(int argc, char const *argv[]) {
-    int maze[size][size] = {
-        {1, 0, 1, 0}, {1, 0, 1, 1}, {1, 0, 0, 1}, {1, 1, 1, 1}};
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    const int size = 4;
+    std::array<std::array<int, size>, size> maze = {
+        std::array<int, size>{1, 0, 1, 0}, std::array<int, size>{1, 0, 1, 1},
+        std::array<int, size>{1, 0, 0, 1}, std::array<int, size>{1, 1, 1, 1}};
 
-    int soln[size][size];
+    std::array<std::array<int, size>, size> soln{};
 
+    // Backtracking: setup matrix solution to zero
     for (int i = 0; i < size; ++i) {
         for (int j = 0; j < size; ++j) {
             soln[i][j] = 0;
         }
     }
 
-    int currposrow = 0;
-    int currposcol = 0;
-    solveMaze(currposrow, currposcol, maze, soln);
+    int currposrow = 0;  // Current position in rows
+    int currposcol = 0;  // Current position in columns
+
+    backtracking::rat_maze::solveMaze<size>(currposrow, currposcol, maze, soln);
```

Comment:
```suggestion
    assert(backtracking::rat_maze::solveMaze<size>(currposrow, currposcol, maze, soln));
```

Reply:
something similar, yes. 
This would require the `backtracking::rat_maze::solveMaze` function to return a boolean type. 
the output of this function should match an expected result.
1. pick an example input data with a known result
2. run the algorithm
3. user `assert` to check the output with the expected result.

Reply:
I think it's fine now. 🙂
Please review it again.